### PR TITLE
[danger] Fix danger on CI - paths & api requests

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixes problems in https://github.com/expo/expo/pull/8307.

# How

- fixes paths 
- wraps API request in try...catch

# Test Plan

- in https://github.com/expo/expo/pull/8307

### CODE REVIEW

File: packages/danger/src/ChangelogChecker.ts [hunk 1]
```diff
@@ -65,7 +65,8 @@ async function runAddChangelogCommandAsync(
   suggestedEntries: PackageChangelogEntry[]
 ): Promise<FixedChangelogEntry[]> {
   for (const entry of suggestedEntries) {
-    await spawnAsync('et', [
+    await (path.join(getExpoRepositoryRootDir(), 'bin', 'expotools'),
```
Comment (tsapeta) [inline packages/danger/src/ChangelogChecker.ts:68]: Missing `spawnAsync`? 😄

File: packages/danger/src/Utils.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export function getPackageChangelogRelativePath(packageName: string): string {
 
 export function getExpoRepositoryRootDir(): string {
   // EXPO_ROOT_DIR is set locally by direnv
-  return process.env.EXPO_ROOT_DIR || join(__dirname, '..');
+  return process.env.EXPO_ROOT_DIR || join(__dirname, '../../..');
```
Comment (tsapeta) [inline packages/danger/src/Utils.ts:15]: Why the path is different than you expected? 🤔

**Review by tsapeta — APPROVED** [review]

File: packages/danger/src/ChangelogChecker.ts (hunk 1 above)
Comment (lukmccall) [reply packages/danger/src/ChangelogChecker.ts]: Coding and participating in the lecture at the same time is, not always the best idea 😅

File: packages/danger/src/Utils.ts (hunk 2 above)
Comment (lukmccall) [reply packages/danger/src/Utils.ts:15]: I moved `danger` to the `packages` and I didn't change this path. 
Locally, we have set `EXPO_ROOT_DIR`. So, I didn't catch this problem earlier.

**Review by tsapeta — APPROVED** [review]

**Pull request merged**